### ISSUE TITLE
InteriorBasis raises IndexError on MeshLine loaded from MSH

### ISSUE THREAD

**Issue by gdmcbain** [kinnala/scikit-fem#178]
Following #175, another problem is that `InteriorBasis(mesh, ElementLineP1())` raises

> IndexError: index 2 is out of bounds for axis 0 with size 2

if `mesh` is loaded from a Gmsh file:

```python
mesh = MeshLine.load(argv[1])
```

Example: [example.tar.gz](https://github.com/kinnala/scikit-fem/files/3155876/example.tar.gz), where:

* `line.geo`:

```
Point(1) = {0, 0, 0};
Point(2) = {1, 0, 0};
Line(1) = {1, 2};
Physical Point("left") = {1};
Physical Point("right") = {2};
Physical Curve("domain") = {1};
```

(The same error is raised without the Physical entities too.)

* `line.py`

```python
mesh = MeshLine.load(argv[1])
element = ElementLineP1()
basis = InteriorBasis(mesh, element)
```

**Comment by gdmcbain** [comment]
The same error is raised with just
```python
mesh.mapping()
```

**Comment by gdmcbain** [comment]
Or `MappingAffine(mesh)`.

**Comment by gdmcbain** [comment]
It looks like the problem is that `mesh.p.shape[0] == 3` so that we've gotten redundant zeros for `mesh.p[1:]`.

Is the error upstream in `meshio` or here in `skfem.Mesh` in `load` or `from_meshio`?

Does the same problem affect two-dimensional meshes?  Gmsh always uses three-dimensional coordinates for nodes and I think these are passed on by `meshio.read`.

**Comment by gdmcbain** [comment]
I had thought to use `Mesh.dim` to strip the trailing coordinates but I see that it's defined a posteriori in terms of `Mesh.p.shape`.

https://github.com/kinnala/scikit-fem/blob/9ff1fd5c76a5b3044df563721c0c80fe6069e651/skfem/mesh/mesh.py#L99-L101

Would it be reasonable to define it as an a priori class variable, `dim = 1` for `MeshLine`, 2 for `Mesh2D` and 3 for `Mesh3D`?

**Comment by gdmcbain** [comment]
Neither can something like
```python
isinstance(cls, MeshLine)
```
be used in `Mesh.from_meshio` since `MeshLine` isn't defined at that point (similarly for `Mesh2D`).

**Comment by gdmcbain** [comment]
Ah: it was just that `MeshLine` doesn't override `Mesh.strip_extra_coordinates` as `Mesh2D` does.

**Comment by kinnala** [comment]
I suppose this can be closed now?

**Comment by gdmcbain** [comment]
Yes, thanks; this was fixed by 9036ffd which was included in #176.
